### ISSUE TITLE
Test fails using Python 3.7 on Debian Sid

### ISSUE THREAD

**Issue by russel** [RPi-Distro/python-gpiozero#666]
Using Python 3.6.6+ on Debian Sid, on GPIOZero e8bf5b500f9f2ea55d06869a86749b9c91691f68 having run "make develop", running py.test passes but delivers a number of deprecation warnings. For example:
```
…/Python_GPIOZero/tests/test_tools.py:173: DeprecationWarning: generator 'smoothed' raised StopIteration
  assert list(smoothed((), 5)) == []
```
On running with Python 3.7.0+ on Debian Sid, these warning become errors, `StopIteration` is raised

**Comment by russel** [comment]
This appears to be related to PEP 479 https://www.python.org/dev/peps/pep-0479/

**Comment by bennuttall** [comment]
Fixed in #670
